### PR TITLE
Add no-new-privileges flag

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note
buildah build now supports --security-opt no-new-privileges flag.
```

### CODE REVIEW

**Comment by rhatdan** [comment]
@edsantiago PTAL

**Comment by rhatdan** [comment]
Related to: https://github.com/containers/podman/pull/16388

**Comment by ashley-cui** [comment]
LGTM

File: pkg/parse/parse.go [hunk 1]
```diff
@@ -928,10 +929,11 @@ func IsolationOption(isolation string) (define.Isolation, error) {
...
 // Valid values for device look like:
-//    '/dev/sdc"
-//    '/dev/sdc:/dev/xvdc"
-//    '/dev/sdc:/dev/xvdc:rwm"
-//    '/dev/sdc:rm"
+//
+//	'/dev/sdc"
+//	'/dev/sdc:/dev/xvdc"
+//	'/dev/sdc:/dev/xvdc:rwm"
+//	'/dev/sdc:rm"
```
Comment (nalind) [inline pkg/parse/parse.go:936]: What's this doing here?

**Comment by nalind** [comment]
Mostly LGTM.  Not sure how the changes to the godoc for pkg/parse.Device() are related.

**Comment by rhatdan** [comment]
I think my editor did that.

**Comment by TomSweeneyRedHat** [comment]
LGTM
@nalind, any objections to pressing the shiny merge button?

**Review by flouthoc — APPROVED** [review]
/lgtm
/approve

**Pull request merged**